aspnetcore/host-and-deploy/health-checks.md

```diff
@@ -0,0 +1,614 @@
+    {
+        // WriteResponse is a delegate used to write the response.
+        ResponseWriter = WriteResponse
+    });
+}
+
+private static Task WriteResponse(HttpContext httpContext, 
+    HealthReport result)
+{
+    httpContext.Response.ContentType = "application/json";
+
+    var json = new JObject(
+        new JProperty("status", result.Status.ToString()),
+        new JProperty("results", new JObject(result.Entries.Select(pair =>
+            new JProperty(pair.Key, new JObject(
+                new JProperty("status", pair.Value.Status.ToString()),
+                new JProperty("description", pair.Value.Description),
+                new JProperty("data", new JObject(pair.Value.Data.Select(
+                    p => new JProperty(p.Key, p.Value))))))))));
+    return httpContext.Response.WriteAsync(
+        json.ToString(Formatting.Indented));
+}
+```
+
+## Database probe
+
+A health check can specify a database query to run as a boolean test to indicate if the database is responding normally.
+
+The sample app uses [BeatPulse](https://github.com/Xabaril/BeatPulse), a health check library for ASP.NET Core apps, to perform a health check on a SQL Server database. BeatPulse executes a `SELECT 1` query against the database to confirm the connection to the database is healthy.
+
+> [!WARNING]
+> When checking a database connection with a query, choose a query that returns quickly. The query approach runs the risk of overloading the database and degrading its performance. In most cases, running a test query isn't necessary. Merely making a successful connection to the database is sufficient. If you find it necessary to run a query, choose a simple SELECT query, such as `SELECT 1`.
+
+In order to use the BeatPulse library, include a package reference to [AspNetCore.HealthChecks.SqlServer](https://www.nuget.org/packages/AspNetCore.HealthChecks.SqlServer/).
+
+Supply a valid database connection string in the *appsettings.json* file of the sample app. The app uses a SQL Server database named `HealthCheckSample`:
+
+[!code-json[](health-checks/samples/2.x/HealthChecksSample/appsettings.json?highlight=3)]
+
+Register health check services with `AddHealthChecks` in `Startup.ConfigureServices`. The sample app calls BeatPulse's `AddSqlServer` method with the database's connection string (*DbHealthStartup.cs*):
+
+[!code-csharp[](health-checks/samples/2.x/HealthChecksSample/DbHealthStartup.cs?name=snippet_ConfigureServices)]
+
+Call Health Check Middleware in the app processing pipeline in `Startup.Configure`:
+
+[!code-csharp[](health-checks/samples/2.x/HealthChecksSample/DbHealthStartup.cs?name=snippet_Configure)]
+
+To run the database probe scenario using the sample app, execute the following command from the project's folder in a command shell:
+
+```console
+dotnet run --scenario db
+```
+
+> [!NOTE]
+> [BeatPulse](https://github.com/Xabaril/BeatPulse) isn't maintained or supported by Microsoft.
+
+## Database context
```

Comment:
/cc @divega 

I'm not sure that `Database context` is a term that we use. I think it would be more straightforward to call this `DbContext` health checks for `Entity Framework Core DbContext`..

Reply:
Nit: We can't use inline code styles in headings.

The rest of the headings at this level use the word "probe," so we could go with that ...

> Entity Framework Core DbContext probe